### PR TITLE
[BugFix] Fix cumulative compaction ut crash

### LINKED ISSUE #10184: [Bug] Cumulative compaction ut crash
### Steps to reproduce the behavior (Required)
19:37:42 [ RUN      ] CumulativeCompactionTest.test_vertical_compact_succeed
19:37:42 =================================================================
19:37:42 ==11659==ERROR: AddressSanitizer: heap-use-after-free on address 0x61100227e1e0 at pc 0x00000628556f bp 0x7fcbfee11950 sp 0x7fcbfee11948
19:37:42 READ of size 8 at 0x61100227e1e0 thread T3 (up_candidates)
19:37:42     #0 0x628556e in __gnu_cxx::__normal_iterator<starrocks::MemTracker**, std::vector<starrocks::MemTracker*, std::allocator<starrocks::MemTracker*> > >::__normal_iterator(starrocks::MemTracker** const&) /usr/include/c++/10.3.0/bits/stl_iterator.h:976
19:37:53     #1 0x627ecbf in std::vector<starrocks::MemTracker*, std::allocator<starrocks::MemTracker*> >::begin() /usr/include/c++/10.3.0/bits/stl_vector.h:812
19:37:53     #2 0x62787da in starrocks::MemTracker::release(long) /root/starrocks/be/src/runtime/mem_tracker.h:189
19:37:53     #3 0xdb1766d in starrocks::DeleterWithMemTracker<starrocks::Tablet>::operator()(starrocks::Tablet*) const /root/starrocks/be/src/runtime/mem_tracker.h:314
19:37:53     #4 0xdb1d178 in std::_Sp_counted_deleter<starrocks::Tablet*, starrocks::DeleterWithMemTracker<starrocks::Tablet>, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>::_M_dispose() /usr/include/c++/10.3.0/bits/shared_ptr_base.h:474
19:37:53     #5 0x63cbb10 in std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release() /usr/include/c++/10.3.0/bits/shared_ptr_base.h:158
19:37:53     #6 0x63c8525 in std::__shared_count<(__gnu_cxx::_Lock_policy)2>::~__shared_count() /usr/include/c++/10.3.0/bits/shared_ptr_base.h:733
19:37:53     #7 0x84a83d1 in std::__shared_ptr<starrocks::Tablet, (__gnu_cxx::_Lock_policy)2>::~__shared_ptr() /usr/include/c++/10.3.0/bits/shared_ptr_base.h:1183
19:37:53     #8 0x84a83ed in std::shared_ptr<starrocks::Tablet>::~shared_ptr() /usr/include/c++/10.3.0/bits/shared_ptr.h:121
19:37:53     #9 0xe42e959 in ~<lambda> /root/starrocks/be/src/storage/compaction_manager.cpp:67
19:37:53     #10 0xe430b09 in _M_destroy /usr/include/c++/10.3.0/bits/std_function.h:176
19:37:53     #11 0xe43085e in _M_manager /usr/include/c++/10.3.0/bits/std_function.h:200
19:37:53     #12 0xe43050f in _M_manager /usr/include/c++/10.3.0/bits/std_function.h:283
19:37:53     #13 0x668fbaf in std::_Function_base::~_Function_base() /usr/include/c++/10.3.0/bits/std_function.h:245
19:37:53     #14 0x837434f in std::function<void ()>::~function() /usr/include/c++/10.3.0/bits/std_function.h:303
19:37:53     #15 0xebb1160 in starrocks::FunctionRunnable::~FunctionRunnable() /root/starrocks/be/src/util/threadpool.cpp:40
19:37:53     #16 0xebb14b1 in void __gnu_cxx::new_allocator<starrocks::FunctionRunnable>::destroy<starrocks::FunctionRunnable>(starrocks::FunctionRunnable*) /usr/include/c++/10.3.0/ext/new_allocator.h:156
19:37:53     #17 0xebb1438 in void std::allocator_traits<std::allocator<starrocks::FunctionRunnable> >::destroy<starrocks::FunctionRunnable>(std::allocator<starrocks::FunctionRunnable>&, starrocks::FunctionRunnable*) /usr/include/c++/10.3.0/bits/alloc_traits.h:531
19:37:53     #18 0xebb1256 in std::_Sp_counted_ptr_inplace<starrocks::FunctionRunnable, std::allocator<starrocks::FunctionRunnable>, (__gnu_cxx::_Lock_policy)2>::_M_dispose() /usr/include/c++/10.3.0/bits/shared_ptr_base.h:560
19:37:53     #19 0x63cbb10 in std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release() /usr/include/c++/10.3.0/bits/shared_ptr_base.h:158
19:37:53     #20 0x63c8525 in std::__shared_count<(__gnu_cxx::_Lock_policy)2>::~__shared_count() /usr/include/c++/10.3.0/bits/shared_ptr_base.h:733
19:37:53     #21 0xd1daee7 in std::__shared_ptr<starrocks::Runnable, (__gnu_cxx::_Lock_policy)2>::~__shared_ptr() /usr/include/c++/10.3.0/bits/shared_ptr_base.h:1183
19:37:53     #22 0xeb9abac in std::__shared_ptr<starrocks::Runnable, (__gnu_cxx::_Lock_policy)2>::reset() /usr/include/c++/10.3.0/bits/shared_ptr_base.h:1301
19:37:53     #23 0xeb936f0 in starrocks::ThreadPool::dispatch_thread() /root/starrocks/be/src/util/threadpool.cpp:523
19:37:53     #24 0xebb0c6d in void std::__invoke_impl<void, void (starrocks::ThreadPool::*&)(), starrocks::ThreadPool*&>(std::__invoke_memfun_deref, void (starrocks::ThreadPool::*&)(), starrocks::ThreadPool*&) /usr/include/c++/10.3.0/bits/invoke.h:73
19:37:53     #25 0xebb05c6 in std::__invoke_result<void (starrocks::ThreadPool::*&)(), starrocks::ThreadPool*&>::type std::__invoke<void (starrocks::ThreadPool::*&)(), starrocks::ThreadPool*&>(void (starrocks::ThreadPool::*&)(), starrocks::ThreadPool*&) /usr/include/c++/10.3.0/bits/invoke.h:95
19:37:53     #26 0xebaf9bd in void std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>::__call<void, , 0ul>(std::tuple<>&&, std::_Index_tuple<0ul>) /usr/include/c++/10.3.0/functional:416
19:37:53     #27 0xebae31f in void std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>::operator()<, void>() /usr/include/c++/10.3.0/functional:499
19:37:53     #28 0xebab383 in void std::__invoke_impl<void, std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>&>(std::__invoke_other, std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>&) /usr/include/c++/10.3.0/bits/invoke.h:60
19:37:53     #29 0xeba8ce7 in std::enable_if<is_invocable_r_v<void, std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>&>, void>::type std::__invoke_r<void, std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>&>(std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>&) /usr/include/c++/10.3.0/bits/invoke.h:110
19:37:53     #30 0xeba491e in std::_Function_handler<void (), std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()> >::_M_invoke(std::_Any_data const&) /usr/include/c++/10.3.0/bits/std_function.h:291
19:37:53     #31 0x931ae7d in std::function<void ()>::operator()() const /usr/include/c++/10.3.0/bits/std_function.h:622
19:37:53     #32 0xeb7f0f0 in starrocks::Thread::supervise_thread(void*) /root/starrocks/be/src/util/thread.cpp:326
19:37:53     #33 0x7fcc064b4ea4 in start_thread (/lib64/libpthread.so.0+0x7ea4)
19:37:53     #34 0x7fcc05acfb0c in clone (/lib64/libc.so.6+0xfeb0c)
19:37:53 
19:37:53 0x61100227e1e0 is located 96 bytes inside of 216-byte region [0x61100227e180,0x61100227e258)
19:37:53 freed by thread T0 here:
19:37:53     #0 0x62365a7 in operator delete(void*, unsigned long) ../../.././libsanitizer/asan/asan_new_delete.cpp:172
19:37:53     #1 0x6289056 in std::default_delete<starrocks::MemTracker>::operator()(starrocks::MemTracker*) const /usr/include/c++/10.3.0/bits/unique_ptr.h:85
19:37:53     #2 0x628072c in std::unique_ptr<starrocks::MemTracker, std::default_delete<starrocks::MemTracker> >::~unique_ptr() /usr/include/c++/10.3.0/bits/unique_ptr.h:361
19:37:53     #3 0x8d84e84 in starrocks::vectorized::CumulativeCompactionTest::~CumulativeCompactionTest() /root/starrocks/be/test/storage/cumulative_compaction_test.cpp:27
19:37:53     #4 0x8d87110 in starrocks::vectorized::CumulativeCompactionTest_test_horizontal_compact_succeed_Test::~CumulativeCompactionTest_test_horizontal_compact_succeed_Test() /root/starrocks/be/test/storage/cumulative_compaction_test.cpp:241
19:37:53     #5 0x8d8712b in starrocks::vectorized::CumulativeCompactionTest_test_horizontal_compact_succeed_Test::~CumulativeCompactionTest_test_horizontal_compact_succeed_Test() /root/starrocks/be/test/storage/cumulative_compaction_test.cpp:241
19:37:53     #6 0x11acd620 in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2433
19:37:53     #7 0x11acd620 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2469
19:37:53     #8 0x11ac40cb in testing::TestInfo::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2690
19:37:53     #9 0x11ac40cb in testing::TestInfo::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2657
19:37:53     #10 0x11ac4214 in testing::TestSuite::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2816
19:37:53     #11 0x11ac4214 in testing::TestSuite::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2795
19:37:53     #12 0x11ac4765 in testing::internal::UnitTestImpl::RunAllTests() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:5338
19:37:53     #13 0x11ac496d in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2433
19:37:53     #14 0x11ac496d in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2469
19:37:53     #15 0x11ac496d in testing::UnitTest::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:4925
19:37:53     #16 0x627bc25 in RUN_ALL_TESTS() (/root/starrocks/be/ut_build_ASAN/test/starrocks_test+0x627bc25)
19:37:53     #17 0x6274847 in main /root/starrocks/be/test/test_main.cpp:75
19:37:53     #18 0x7fcc059f3554 in __libc_start_main (/lib64/libc.so.6+0x22554)
19:37:53 
19:37:53 previously allocated by thread T0 here:
19:37:53     #0 0x6235787 in operator new(unsigned long) ../../.././libsanitizer/asan/asan_new_delete.cpp:99
19:37:53     #1 0x8d84329 in starrocks::vectorized::CumulativeCompactionTest::SetUp() /root/starrocks/be/test/storage/cumulative_compaction_test.cpp:200
19:37:53     #2 0x11acd620 in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2433
19:37:53     #3 0x11acd620 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2469
19:37:53     #4 0x11ac3f38 in testing::Test::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2503
19:37:53     #5 0x11ac3f38 in testing::Test::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2498
19:37:53     #6 0x11ac4124 in testing::TestInfo::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2684
19:37:53     #7 0x11ac4124 in testing::TestInfo::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2657
19:37:53     #8 0x11ac4214 in testing::TestSuite::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2816
19:37:53     #9 0x11ac4214 in testing::TestSuite::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2795
19:37:53     #10 0x11ac4765 in testing::internal::UnitTestImpl::RunAllTests() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:5338
19:37:53     #11 0x11ac496d in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2433
19:37:53     #12 0x11ac496d in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:2469
19:37:53     #13 0x11ac496d in testing::UnitTest::Run() /var/local/thirdparty/src/googletest-release-1.10.0/googletest/src/gtest.cc:4925
19:37:53     #14 0x627bc25 in RUN_ALL_TESTS() (/root/starrocks/be/ut_build_ASAN/test/starrocks_test+0x627bc25)
19:37:53     #15 0x6274847 in main /root/starrocks/be/test/test_main.cpp:75
19:37:53     #16 0x7fcc059f3554 in __libc_start_main (/lib64/libc.so.6+0x22554)
19:37:53 
19:37:53 Thread T3 (up_candidates) created by T0 here:
19:37:53     #0 0x61df482 in __interceptor_pthread_create ../../.././libsanitizer/asan/asan_interceptors.cpp:214
19:37:53     #1 0xeb7e544 in starrocks::Thread::start_thread(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const&, unsigned long, scoped_refptr<starrocks::Thread>*) /root/starrocks/be/src/util/thread.cpp:281
19:37:53     #2 0xeb9adb4 in starrocks::Status starrocks::Thread::create<void (starrocks::ThreadPool::*)(), starrocks::ThreadPool*>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, void (starrocks::ThreadPool::* const&)(), starrocks::ThreadPool* const&, scoped_refptr<starrocks::Thread>*) /root/starrocks/be/src/util/thread.h:55
19:37:53     #3 0xeb94941 in starrocks::ThreadPool::create_thread() /root/starrocks/be/src/util/threadpool.cpp:565
19:37:53     #4 0xeb8ebbc in starrocks::ThreadPool::init() /root/starrocks/be/src/util/threadpool.cpp:254
19:37:53     #5 0xeb8b861 in starrocks::ThreadPoolBuilder::build(std::unique_ptr<starrocks::ThreadPool, std::default_delete<starrocks::ThreadPool> >*) const /root/starrocks/be/src/util/threadpool.cpp:81
19:37:53     #6 0xe42d615 in starrocks::CompactionManager::CompactionManager() /root/starrocks/be/src/storage/compaction_manager.cpp:16
19:37:53     #7 0xda6e032 in starrocks::StorageEngine::StorageEngine(starrocks::EngineOptions const&) /root/starrocks/be/src/storage/storage_engine.cpp:95
19:37:54     #8 0xda90896 in std::_MakeUniq<starrocks::StorageEngine>::__single_object std::make_unique<starrocks::StorageEngine, starrocks::EngineOptions const&>(starrocks::EngineOptions const&) /usr/include/c++/10.3.0/bits/unique_ptr.h:962
19:37:54     #9 0xda6d33e in starrocks::StorageEngine::open(starrocks::EngineOptions const&, starrocks::StorageEngine**) /root/starrocks/be/src/storage/storage_engine.cpp:78
19:37:54     #10 0x62745e5 in main /root/starrocks/be/test/test_main.cpp:64
19:37:54     #11 0x7fcc059f3554 in __libc_start_main (/lib64/libc.so.6+0x22554)
19:37:54 
19:37:54 SUMMARY: AddressSanitizer: heap-use-after-free /usr/include/c++/10.3.0/bits/stl_iterator.h:976 in __gnu_cxx::__normal_iterator<starrocks::MemTracker**, std::vector<starrocks::MemTracker*, std::allocator<starrocks::MemTracker*> > >::__normal_iterator(starrocks::MemTracker** const&)
19:37:54 Shadow bytes around the buggy address:
19:37:54   0x0c2280447be0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
19:37:54   0x0c2280447bf0: 00 00 00 00 00 00 00 00 00 00 fa fa fa fa fa fa
19:37:54   0x0c2280447c00: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
19:37:54   0x0c2280447c10: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
19:37:54   0x0c2280447c20: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
19:37:54 =>0x0c2280447c30: fd fd fd fd fd fd fd fd fd fd fd fd[fd]fd fd fd
19:37:54   0x0c2280447c40: fd fd fd fd fd fd fd fd fd fd fd fa fa fa fa fa
19:37:54   0x0c2280447c50: fa fa fa fa fa fa fa fa 00 00 00 00 00 00 00 00
19:37:54   0x0c2280447c60: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
19:37:54   0x0c2280447c70: 00 00 fa fa fa fa fa fa fa fa fa fa fa fa fa fa
19:37:54   0x0c2280447c80: fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd
19:37:54 Shadow byte legend (one shadow byte represents 8 application bytes):
19:37:54   Addressable:           00
19:37:54   Partially addressable: 01 02 03 04 05 06 07 
19:37:54   Heap left redzone:       fa
19:37:54   Freed heap region:       fd
19:37:54   Stack left redzone:      f1
19:37:54   Stack mid redzone:       f2
19:37:54   Stack right redzone:     f3
19:37:54   Stack after return:      f5
19:37:54   Stack use after scope:   f8
19:37:54   Global redzone:          f9
19:37:54   Global init order:       f6
19:37:54   Poisoned by user:        f7
19:37:54   Container overflow:      fc
19:37:54   Array cookie:            ac
19:37:54   Intra object redzone:    bb
19:37:54   ASan internal:           fe
19:37:54   Left alloca redzone:     ca
19:37:54   Right alloca redzone:    cb
19:37:54   Shadow gap:              cc
19:37:54 ==11659==ABORTING
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10184

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/test/storage/cumulative_compaction_test.cpp [hunk 1]
```diff
@@ -26,6 +26,13 @@ static StorageEngine* k_engine = nullptr;
```
Comment (rickif) [inline be/test/storage/cumulative_compaction_test.cpp:26]: How about changing `k_engine` as a member of `CumulativeCompactionTest`?

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — DISMISSED** [review]

**Review by imay — DISMISSED** [review]

**Comment by meegoo** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**